src/bulk/common.ts

```diff
@@ -588,38 +546,61 @@ function executeCommands(
           ? new DeleteOperation(bulkOperation.s.namespace, batch.operations, finalOptions)
           : null;
 
-    if (operation != null) {
-      executeOperation(bulkOperation.s.collection.client, operation).then(
-        result => resultHandler(undefined, result),
-        error => resultHandler(error)
-      );
+    if (operation == null) throw new MongoRuntimeError(`Unknown batchType: ${batch.batchType}`);
```

Comment:
TS narrowing 🤷🏻 what can ya do

Reply:
I'm not sure I follow.  What problematic narrowing are you referring to here? 

On line 347 we explicitly assign to `null`, so we need to narrow.  Alternatively, the ternary could just throw if an unexpected batch type was encountered (MongoRuntimeError - should never happen).

Reply:
Yea you got it right, I just called it out for the reviewer that is wondering why it is necessary. 

I can't make the ternary throw (yet!) so this is the best we got.